addons/wiki/models.py

```diff
@@ -251,3 +256,80 @@ def clone_wiki_versions(cls, node, copy, user, save=True):
         if save:
             copy.save()
         return copy
+
+
+class NodeSettings(BaseNodeSettings):
+    complete = True
+    has_auth = True
+    is_publicly_editable = models.BooleanField(default=False, db_index=True)
+
+    def set_editing(self, permissions, auth=None, log=False):
+        """Set the editing permissions for this node.
+
+        :param auth: All the auth information including user, API key
+        :param bool permissions: True = publicly editable
+        :param bool save: Whether to save the privacy change
+        :param bool log: Whether to add a NodeLog for the privacy change
+            if true the node object is also saved
+        """
+        node = self.owner
+
+        if permissions and not self.is_publicly_editable:
+            if node.is_public:
+                self.is_publicly_editable = True
+            else:
+                raise NodeStateError('Private components cannot be made publicly editable.')
+        elif not permissions and self.is_publicly_editable:
+            self.is_publicly_editable = False
+        else:
+            raise NodeStateError('Desired permission change is the same as current setting.')
+
+        if log:
+            node.add_log(
+                action=(NodeLog.MADE_WIKI_PUBLIC
+                        if self.is_publicly_editable
+                        else NodeLog.MADE_WIKI_PRIVATE),
+                params={
+                    'project': node.parent_id,
+                    'node': node._primary_key,
+                },
+                auth=auth,
+                save=False,
+            )
+            node.save()
```

Comment:
Not sure why you're passing `save=False` which tells `add_log` to not save the Node then calling `node.save()` on the next line.